src/VirtoCommerce.MarketingModule.Data/Repositories/MarketingRepository.cs

```diff
@@ -177,6 +177,16 @@ public Task RemoveMarketingUsagesAsync(string[] ids)
         }
         #endregion
 
+        public Task<string[]> CheckCouponsForUniqueness(Coupon[] coupons)
```

Comment:
What if it would be 1000 coupons?
Maybe we should perform the action in batches.

Reply:
Done